include/kernel.h

```diff
@@ -4452,6 +4452,25 @@ void k_heap_init(struct k_heap *h, void *mem, size_t bytes);
  */
 void *k_heap_alloc(struct k_heap *h, size_t bytes, k_timeout_t timeout);
 
+/** @brief Allocate aligned memory from a k_heap
+ *
+ * Behaves in all ways like k_heap_alloc(), except that the returned
+ * memory (if available) will have a starting address in memory which
+ * is a multiple of the specified power-of-two alignment value in
+ * bytes.  The resulting memory can be returned to the heap using
+ * k_heap_free().
+ *
+ * @note Can be called by ISRs, but @a timeout must be set to K_NO_WAIT.
+ *
+ * @param h Heap from which to allocate
+ * @param align Alignment in bytes, must be a power of two
+ * @param bytes Number of bytes requested
+ * @param timeout How long to wait, or K_NO_WAIT
+ * @return Pointer to memory the caller can now use
+ */
+void *k_heap_aligned_alloc(struct k_heap *h, size_t align, size_t bytes,
```

Comment:
Just a thought, since this is still open.

Would it make more sense to make this `static inline` since it is really just a small wrapper?

Reply:
I can change it accordingly

Reply:
I should say, make the non-aligned version a `static inline` wrapper - should be obvious - sorry for the confusion ;-)

Reply:
Yes, I already suspected that ;)